airflow/jobs/scheduler_job.py

```diff
@@ -1748,6 +1748,8 @@ def _schedule_dag_run(
     def _verify_integrity_if_dag_changed(self, dag_run: DagRun, session=None):
         """Only run DagRun.verify integrity if Serialized DAG has changed since it is slow"""
         latest_version = SerializedDagModel.get_latest_version_hash(dag_run.dag_id, session=session)
+        if latest_version is None:
+            raise AirflowException(f"Serialized DAG not found for DAG run {dag_run.id}.")
```

Comment:
What does raising an exception here do to the scheduler loop?

Reply:
```suggestion
            raise SerializedDagNotFound(f"Serialized DAG not found for DAG run {dag_run.id}.")
```